specification/src/main/asciidoc/platform/ApplicationProgrammingInterface.adoc

```diff
@@ -1000,91 +1000,48 @@ _https://jakarta.ee/specifications/xml-web-services/_ .
-
-The Jakarta EE security restrictions typically
-prevent all application component types, except application clients,
-from creating and exporting an RMI-IIOP object. All Jakarta EE application
-component types can be clients of RMI-IIOP objects. Jakarta EE applications
-should also use JNDI to lookup non-Enterprise Beans RMI-IIOP objects. The JNDI names
-used for such non-Enterprise Beans RMI-IIOP objects should be configured at
-deployment time using the standard environment entries mechanism (see
-<<a607, JNDI Naming Context>>).
-The application should fetch a name from JNDI using an environment
-entry, and use the name to lookup the RMI-IIOP object. Typically such
-names will be configured to be names in the COSNaming name service.
-
-This specification does not provide a
-portable way for applications to bind objects to names in a name
-service. Some products may support use of JNDI and COSNaming for binding
-objects, but this is not required. Portable Jakarta EE application clients
-can create non-Enterprise Beans RMI-IIOP server objects for use as callback objects,
-or to pass in calls to other RMI-IIOP objects.
-
-Note that while RMI-IIOP doesn’t specify how
-to propagate the current security context or transaction context, the
-Jakarta Enterprise Beans interoperability specification does define such context propagation.
-This specification only requires that the propagation of context
-information as defined in the Jakarta Enterprise Beans specification be supported in the use
-of RMI-IIOP to access enterprise beans. The propagation of context
-information is not required in the uses of RMI-IIOP to access objects
-other than enterprise beans.
-
-The RMI-IIOP specification describes how
-portable Stub and _Tie_ classes can be created. To be portable to all
-implementations that use a CORBA Portable Object Adapter (POA), the
-_Tie_ classes must extend the _org.omg.PortableServer.Servant_ class.
-This is typically done by using the _-poa_ option to the _rmic_ command.
-A Jakarta EE product must provide support for these portable _Stub_ and
-_Tie_ classes, typically using the required CORBA POA. However, for
-portability to systems that do not use a POA to implement RMI-IIOP,
-applications should not depend on the fact that the _Tie_ extends the
-_Servant_ class. A Jakarta EE application that defines or uses RMI-IIOP
-objects other than enterprise beans must include such portable _Stub_
-and _Tie_ classes in the application package. _Stub_ and _Tie_ objects
-for enterprise beans, however, must not be included with the
-application: they will be generated, if needed, by the Jakarta EE product
-at deployment time or at run time.
-
-RMI-IIOP is defined by chapters 5, 6, 13, 15,
-and section 10.6.2 of the CORBA 2.3.1 specification, available at
-_http://www.omg.org/cgi-bin/doc?formal/99-10-07_ , and by the Java™
-Language To IDL Mapping Specification, available at
-_http://www.omg.org/cgi-bin/doc?ptc/2000-01-06_ .
+===== RMI-IIOP
```

Comment:
Rewriting my original comment on this section, so deleted my previous comment rather than updating, not sure if this is clearer or not:

As per https://openjdk.java.net/jeps/320, `javax.rmi.PortableRemoteObject` is removed from Java SE 11 (which is indeed part of Corba/RMI-IIOP).

Whether we drop or rename or keep the RMI-IIOP section, could we refer to the EJB spec dependency on `javax.rmi.PortableRemoteObject` as `legacy` support?  Using the `optional` word to describe the EJB dependency on `javax.rmi.PortableRemoteObject` implies too much (e.g. having javax.rmi.PortableRemoteObject using TCK tests that only run on Java SE 8).